wgpu-hal/src/vulkan/device.rs

```diff
@@ -377,7 +377,11 @@ impl
             })
             .collect::<ArrayVec<_, 8>>();
 
-        let mut vk_flags = vk::DescriptorPoolCreateFlags::empty();
+        let mut vk_flags = if self.private_caps.update_after_bind {
+            vk::DescriptorPoolCreateFlags::UPDATE_AFTER_BIND
```

Comment:
we shouldn't use `private_caps` here. Instead, the incoming `flags` should be checked for this bit.

Reply:
Still an issue? We should check for `flags` here, not `self.uab_types`